### PR TITLE
[Spike][Docs] reorganize connector reference into reference, how-to guides, and tutorials

### LINKED ISSUE #5961: [Spike] docs: untangle connector reference into reference, how-to guides and tutorials
Originated from: https://github.com/strongloop/loopback-next/issues/5786

Part of #5960 

## Acceptance Criteria
- [ ] Review the existing connector reference contents and create a plan on how to extract out content into reference, how-to guides and tutorials. 
   - As the outcome of this spike, once agreed upon, follow up stories will be created under epic #5960.

### PR DESCRIPTION
**Pull request by agnes512**
Closes #5961 

## Suggestion

Currently, our connector reference sits under "Reference Guides". It's hard to find a specific connector page as the path might be `Reference Guides/Connector reference/Database Connector/MySQL connector/Tutorial`.
Here is a proposal of how to reorganize them into the four quadrants 

## REFERENCE GUIDES
keep the current connector reference as is.

## BEHIND THE SCENES
image

add a section or a subpage under "Behind the scene/Datasource" that talk about:
- what is connectors
- what role it plays in req/res cycle, and what relation it has between model and other artifacts (briefly)
- connector types (add links ):
  - SQL (transaction), NoSQL (freeform properties)
  - Services connectors (service proxy) and others
  - (optional) In-memory connector
  - (optional) Community connector

## HOW-TOS

Screen Shot 2020-08-05 at 5 04 01 PM

add a page "Configuring DataSource"(or a better name) under "How-tos/Access Database" that talks about common connector configurations such as:
- how to configure the connector ( what properties the connector configuration might take
- how type maps ⚠️ should finish task Add missing LoopBack 4 Types docs #5063 first
  - to database (Database Migration), e.g String -> `Varchar, Boolean -> Tinyint`
  - from database (Discovery), e.g `Bigint -> int, Text -> String`
- how to customize table/column names
- how to discover/migration tables/models
- list connector specific topics, e.g transaction, `objectId`, and readers should check the corresponding connector pages.
- add links to connector reference

## TUTORIALS
Screen Shot 2020-08-05 at 5 11 19 PM

Add a page "Connect to back-end tutorial" under "Tutorials" that:
- Briefly introduce what is connectors and what connectors LB provides. 
- List the connector tutorials we have in connector ref (MySQL, PostgreSQL, Oracle, and MongoDB)
- Add the link to connector ref

### CODE REVIEW

**Comment by jannyHou** [comment]
- "a section or a subpage under **Behind the scene/Datasource** " 👍 
- "add a page "Configuring DataSource"(or a better name) under **How-tos/Access Database**" 👍 
- "add a page **Connect to back-end tutorial** under **Tutorials**" 👍 

The plan LGTM, awesome.

File: docs/site/Configuring-datasource.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+lang: en
+title: 'Configuring DataSource'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, Connector, Datasource
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Configuring-datasource.html
+---
+
+## Overview
```
Comment (bajtos) [inline docs/site/Configuring-datasource.md:9]: Please add a short note to explain readers that this is a placeholder page and add a link to the GH issue that's tracking the task of writing the actual content. See https://github.com/strongloop/loopback-next/blob/master/docs/site/migration/clients.md for inspiration.

```
{% include note.html content="
This is a placeholder page, the task of adding content is tracked by the following GitHub issue: 
[loopback-next#3954](https://github.com/strongloop/loopback-next/issues/3954)
" %}
```

File: docs/site/DataSources.md [hunk 2]
```diff
@@ -81,3 +81,17 @@ Attach the newly created datasource to the app by calling `app.dataSource()`.
 The `app.datasource()` method is available only on application classes
 with `RepositoryMixin` applied.
 " %}
+
```
Comment (bajtos) [inline docs/site/DataSources.md:84]: I think the content above (e.g. "Creating a datasource at runtime") should be eventually moved to "How-to guides" and/or the new page "Configuring-datasource.md" you are creating? I guess that's out of scope of this PR, just want to mention it.

File: docs/site/DataSources.md [hunk 3]
```diff
@@ -81,3 +81,17 @@ Attach the newly created datasource to the app by calling `app.dataSource()`.
...
+### Connector
+
+<!-- TODO
```
Comment (bajtos) [inline docs/site/DataSources.md:87]: Please add a short note to explain readers that this is a placeholder page and add a link to the GH issue that's tracking the task of writing the actual content.

File: docs/site/tutorials/connectors/Connecting-to-back-end.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
+lang: en
+title: 'Connecting to Back-end Service Tutorials'
+keywords:
+  LoopBack 4.0, LoopBack 4, Node.js, TypeScript, Tutorial, Connector, Datasource
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Connecting-to-back-end.html
+---
+
+<!-- TODO
```
Comment (bajtos) [inline docs/site/tutorials/connectors/Connecting-to-back-end.md:10]: Please add a short note to explain readers that this is a placeholder page and add a link to the GH issue that's tracking the task of writing the actual content.

**Pull request merged**